File: 45-jump-game-ii/jump-game-ii.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        cnt = 0
        prev = 0
        max_num = 0
        for i in range(len(nums)-1):
            max_num = max(max_num, i + nums[i])

            if (prev == i):
                prev = max_num
                cnt += 1
        return cnt

File: 45-jump-game-ii/jump-game-ii.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var jump = function(nums) {
    let cnt = 0;
    let prev = 0;
    let max = 0;
    for (let i = 0; i < nums.length - 1; i++) {
        max = Math.max(max, i + nums[i]);
        if (i === prev) {
            cnt++;
            prev = max;
        }
    }
    return cnt;
};